### PR TITLE
Enable xla build caching

### PR DESCRIPTION
**Pull request by yeounoh**
### Description

Enable cloud caching for xla build.

### Issue

Addresses #78182 

### Testing

The change/caching was tested in the downstream [workflow](https://app.circleci.com/pipelines/github/pytorch/xla/12306/workflows/ca613a3c-de81-40ad-8498-72ebc65d42f6/jobs/27211)

### CODE REVIEW

File: .jenkins/pytorch/common_utils.sh [hunk 1]
```diff
@@ -114,10 +114,11 @@ function checkout_install_torchvision() {
 function clone_pytorch_xla() {
   if [[ ! -d ./xla ]]; then
     git clone --recursive --quiet https://github.com/pytorch/xla.git
-    pushd xla
-    # pin the xla hash so that we don't get broken by changes to xla
-    git checkout "$(cat ../.github/ci_commit_pins/xla.txt)"
-    popd
+    # Test caching with the latest master
```
Comment (yeounoh) [inline .jenkins/pytorch/common_utils.sh:117]: We will re-enable the pin after testing cc @clee2000

File: setup.py [hunk 2]
```diff
@@ -1107,6 +1107,7 @@ def print_box(msg):
                 'include/torch/csrc/lazy/core/*.h',
                 'include/torch/csrc/lazy/core/internal_ops/*.h',
                 'include/torch/csrc/lazy/core/ops/*.h',
+                'include/torch/csrc/lazy/ts_backend/*.h',
```
Comment (yeounoh) [inline setup.py:1110]: Remove this (or rebase) after landing #82384

**Comment by yeounoh** [comment]
Hi @clee2000 , the xla test suceeded in 1.5 hrs with caching. https://github.com/pytorch/pytorch/runs/7553918384?check_suite_focus=true

I will remove the test stubs, so you can review and merge this.

File: .jenkins/pytorch/common_utils.sh (hunk 1 above)
Comment (yeounoh) [reply .jenkins/pytorch/common_utils.sh]: Re-enabled.

**Comment by clee2000** [comment]
would it also be possible to bump the xla hash in this pr to something past the "Enable sccache & remote caching" commit on pytorch/xla? automatic hash updates for xla have been failing recently due to failing jobs and i think the pin is a bit behind

**Comment by yeounoh** [comment]
> would it also be possible to bump the xla hash in this pr to something past the "Enable sccache & remote caching" commit on pytorch/xla? automatic hash updates for xla have been failing recently due to failing jobs and i think the pin is a bit behind

Good catch, done.

**Comment by yeounoh** [comment]
@clee2000 , https://github.com/pytorch/pytorch/pull/82355 has broken PyTorch/XLA -- we have a patch out, I will update the XLA hash again once we merge it.

**Comment by yeounoh** [comment]
It passed with a new XLA hash in 1h 40m. @clee2000 Can we merge this?

**Review by clee2000 — APPROVED** [review]

**Pull request closed without merging**